parse_lib.py

```diff
@@ -95,8 +94,9 @@ def resolve_relative_resource_urls(html_string):
     imgs = html.find_all("img", src=True)
     equations = html.find_all("object", data=True)
     list(map(update_anchor_href, anchors))
```

Comment:
Also, agreed---we can't avoid side-effects, but we should make them as apparent as we can.